source/dem/pw_linear_force.cc

```diff
@@ -95,48 +143,39 @@ PWLinearForce<dim>::calculate_pw_contact_force(
 template <int dim>
 std::tuple<Tensor<1, dim>, Tensor<1, dim>, Tensor<1, dim>, Tensor<1, dim>>
 PWLinearForce<dim>::calculate_linear_contact_force_and_torque(
-  const Parameters::Lagrangian::PhysicalProperties &physical_properties,
-  pw_contact_info_struct<dim> &                     contact_info,
-  const ArrayView<const double> &                   particle_properties)
+  pw_contact_info_struct<dim> &  contact_info,
+  const ArrayView<const double> &particle_properties)
 {
-  // Calculation of effective Young's modulus of the
-  // contact
-  double effective_youngs_modulus =
-    (physical_properties.youngs_modulus_particle *
-     physical_properties.youngs_modulus_wall) /
-    (physical_properties.youngs_modulus_wall *
-       (1 - physical_properties.poisson_ratio_particle *
-              physical_properties.poisson_ratio_particle) +
-     physical_properties.youngs_modulus_particle *
-       (1 - physical_properties.poisson_ratio_wall *
-              physical_properties.poisson_ratio_wall));
+  const unsigned int particle_type =
+    particle_properties[DEM::PropertiesIndex::type];
 
   // Calculation of normal and tangential spring and dashpot constants
   // using particle properties
   double normal_spring_constant =
     1.0667 * sqrt((particle_properties[DEM::PropertiesIndex::dp] / 2)) *
-    effective_youngs_modulus *
+    this->effective_youngs_modulus[particle_type] *
```

Comment:
Gather it once and store it into a double variable. Will make everything easier to read.

Reply:
Can you please check if this comment is addressed? I am not sure I understood the comment.